Generals/Code/CMakeLists.txt

```diff
@@ -24,8 +24,10 @@ target_include_directories(gi_libraries_source_wwvegas_wwmath INTERFACE "Librari
 target_include_directories(gi_libraries_source_wwvegas_wwsaveload INTERFACE "Libraries/Source/WWVegas/WWSaveLoad")
 target_include_directories(gi_main INTERFACE "Main")
 target_link_libraries(gi_always INTERFACE
-    gi_libraries_include
     core_utility
+    gi_libraries_include
+    # Must stay below so headers from game are included first
```

Comment:
This order dependency will not be necessary if the core's BaseType.h is put into a different folder: "Adapter/Lib/BaseType.h".

Reply:
I think there is a problem with my proposed "Adapter/Lib/BaseType.h" approach:

What if a Generals or GeneralsMD file includes a Core header that includes "Lib/BaseType.h"? Then it will not include the correct file right?

If that is the case, then your BaseTypeCore.h renaming proposal is the better way.